README.md

```diff
@@ -4,222 +4,1008 @@
+    * Bash completion support.
 
-This feature is still in the preliminary phase, so may not be *quite* correct or useable, but it's better than nothing.
+    * Coloured, configurable logging.
 
-Basically, build and install your tool into its final resting place on your machine, then run `myTool __jcli:bash_complete_script` to make it output
-a bash completion script. Further instructions completely depend on how your system is set up.
+    * User Input that integrates with User-Defined argument binding and validation.
 
-For example, systems with bash_completion.d can run a command similar to (at least, I think) `myTool __jcli:bash_complete_script > /etc/bash_completion.d/myTool`
-to automatically enable it completion. You can [see an example](https://bradley.chatha.dev/BlogPost/JcliNews/4-v0-9-0-release-changelog-jcli-dlang-library#bash-completion) over on my blog.
+    * Decent support for writing and parsing ANSI text.
 
-Another example could be `eval "$(myTool __jcli:bash_complete_script)"`, which will enable completion for the current shell session.
+    * Basic but flexible Configuration Providers, used alongside Dependency Injection.
 
-The resulting bash script will essentially generate a call to `myTool __jcli:complete COMP_CWORD COMP_WORDS`, where JCLI will then do the actual busywork.
+    * An ANSI-enabled text buffer, for easier and efficient control over coloured, non-uniform text output.
 
-I'm still a pretty big noob to Linux, and this feature is still in its early stages, so forgive me if things aren't exactly perfect here :o)
+    * Shell utilities such as `pushLocation` and `popLocation`, synonymous with Powershell's `Push-Location` and `Pop-Location`.
 
-## Examples
+* Uncategorised:
 
-There are documented examples within the [examples](https://github.com/SealabJaster/jcli/tree/master/examples) folder.
+    * All individual parts of this library are intended to be reusable. Allowing you to build your own CLI core using these already-made components, if desired.
 
-Other than there, here's a list of projects/scripts using JCLI:
+# Quick Start
 
-* [JCLI's test runner](https://github.com/SealabJaster/jcli/blob/master/examples/test.d)
+*This is a brief overview, for slightly more in-depth examples please look at the fully-documented [examples](https://github.com/BradleyChatha/jcli/tree/master/examples) folder.*
 
-* [JCLI's manual/misc test program](https://github.com/SealabJaster/jcli_testerr)
+## Creating a default command
 
-## Changelog
+The default command is the command that is ran when you don't specify the name of a named command. e.g. `mytool 60 20 --some=args` would call the default command if it exists:
 
-I publish changelogs over at [my website](https://bradley.chatha.dev/Blog/JcliNews).
+```d
+// inside of app.d
+module app;
+import jaster.cli;
 
-## Quick Start
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    int onExecute()
```

Comment:
I didn't consider about using `opCall`. I did however consider using a UDA.

I fear this is mostly a bikeshedding topic though, since it's more of a user-facing design choice rather than a technical one.

I went with `onExecute` so that every command has the same name for its executor function, which while it's technically less flexible than a UDA, I don't think there's really any value (especially at this point) in letting the user use whatever name they want.

It also ties in slightly with regards to inheritance, as seen in the "inheritance" section.

It might not be the best name either (e.g. `executeCommand`, or just `execute` could've been used), but I hope it's descriptive enough.